node/bindnode/repr.go

```diff
@@ -605,13 +618,25 @@ func (w *_assemblerRepr) AssignInt(i int64) error {
 	case schema.UnionRepresentation_Kinded:
 		return w.asKinded(stg, datamodel.Kind_Int).AssignInt(i)
 	case schema.EnumRepresentation_Int:
-		typ := w.schemaType.(*schema.TypeEnum)
-		for _, member := range typ.Members() {
-			if int64(stg[member]) == i {
+		for member, reprInt := range stg {
+			if int64(reprInt) != i {
+				continue
+			}
+			val := (*_assembler)(w).nonPtrVal()
+			if val.Kind() == reflect.String {
 				return (*_assembler)(w).AssignString(member)
+			} else {
+				// Short-cut to storing the repr int directly, akin to node.go's AssignInt.
+				val.SetInt(i)
```

Comment:
Are we okay with using an else here?  Maybe we should prefer to use a case switch?

As written with an "else" like this, if one has a golang `bool` field, we'll still go down this branch, and get a panic from SetInt and it won't tell the user anything useful about their schema-vs-struct mismatch.